src/EditorFeatures/Test/Structure/StructureTaggerTests.cs

```diff
@@ -121,22 +94,31 @@ End Module
             using var workspace = TestWorkspace.CreateVisualBasic(code, composition: EditorTestCompositions.EditorFeaturesWpf);
             var tags = await GetTagsFromWorkspaceAsync(workspace);
 
-            // ensure all 4 outlining region tags were found
-            Assert.Equal(4, tags.Count);
-
-            // ensure only the method outlining region is marked as an implementation
-            Assert.False(tags[0].IsImplementation);
-            Assert.False(tags[1].IsImplementation);
-            Assert.False(tags[2].IsImplementation);
-            Assert.True(tags[3].IsImplementation);
-
-            // verify line counts
-            var hints = tags.Select(x => x.CollapsedHintForm).Cast<ViewHostingControl>().Select(vhc => vhc.TextView_TestOnly).ToList();
-            Assert.Equal(9, hints[0].TextSnapshot.LineCount); // namespace
-            Assert.Equal(7, hints[1].TextSnapshot.LineCount); // region
-            Assert.Equal(5, hints[2].TextSnapshot.LineCount); // class
-            Assert.Equal(3, hints[3].TextSnapshot.LineCount); // method
-            hints.Do(v => v.Close());
+            Assert.Collection(tags,
+                namespaceTag =>
+                {
+                    Assert.False(namespaceTag.IsImplementation);
+                    Assert.Equal(9, GetCollapsedHintLineCount(namespaceTag));
```

Comment:
I'm currently not asserting anything about HeaderSpans for these -- it looks like this is broken and unsure if this is just never worked.

Reply:
Added a commit to attempt to fix some VB behavior.